coding-test/minjae/sort/두배열의원소교체.js

```diff
@@ -0,0 +1,26 @@
+const readline = require("readline"); 
+const rl = readline.createInterface({ 
+    input: process.stdin, 
+    output: process.stdout, 
+}); 
+
+const input = [];
+
+rl.on("line", function (line) {
+    input.push(line); 
+}).on("close", function () { 
+    //교재에 나온 풀이 말고 다른 방법으로 풀어봄 :)
+    const [n, k] = input.shift().split(" ").map(el => parseInt(el));
+    const listOne = input[0].split(" ").map(el => parseInt(el));
+    const listTwo = input[1].split(" ").map(el => parseInt(el));
+
+    let [minOne, maxTwo, minIndex, maxIndex] = [0, 0, 0, 0];
+
+    for(let i = 0; i < k; i++) {
+        [minOne, maxTwo] = [Math.min(...listOne), Math.max(...listTwo)];
+        [minIndex, maxIndex] = [listOne.indexOf(minOne), listTwo.indexOf(maxTwo)];
+        
+        [listOne[minIndex], listTwo[maxIndex]] = [listTwo[maxIndex], listOne[minIndex]];
+    }
```

Comment:
해당 알고리즘은 잘못된 알고리즘입니다.
최대 K번의 교체가 이루어지는 것이 포인트입니다.
무조건 listOne의 합이 최댓값을 갖도록 만들어주어야합니다.
애초에 listOne의 원소들이 listTwo의 원소들보다 크다면 한 번의 교체도 이루어지면 안되는 것이죠.

극단적인 예로,
listOne = [5, 5, 5, 5, 5];
listTwo = [1, 1, 1, 1, 1];
이라고 할 때, 정답은 25이지만 형님 알고리즘의 답은 21이 도출됩니다.

따라서, listOne의 가장 작은 원소가 listTwo의 가장 큰 원소보다 클 때만 교체해주는 비교 조건이 포함되어야 합니다.
아래 처럼 말이죠.

(알고리즘 속도 또한 아래 것이 훨씬 빠르게 동작합니다.)
```suggestion
    listOne.sort((a, b) => a - b);
    listTwo.sort((a, b) => b - a);

    for (let i = 0; i < K; i++) {
        if (listOne[i] < listTwo[i]) {
            [listOne[i], listTwo[i]] = [listTwo[i], listOne[i]];
        } else {
            break;
        }
}
```